backend/notification/rate_limiter.js

```diff
@@ -1,34 +1,119 @@
+const prisma = require("../prisma/prisma_client");
+
 class RateLimiter {
-  constructor(time_window, maximum_requests) {
-    this.time_window = time_window * 1000;
-    this.maximum_requests = maximum_requests;
+  constructor(timeWindow, maximumRequests) {
+    this.timeWindow = timeWindow * 1000;
+    this.maximumRequests = maximumRequests;
     this.userRequestMap = new Map();
   }
 
-  useRateLimiter(userId) {
+  async saveNotificationInteraction(userId, category, notificationId) {
+    await prisma.notificationInteractions.create({
+      data: {
+        userId: userId,
+        timestamp: new Date(),
+        category: category,
+        notificationId: notificationId,
+      },
+    });
+  }
+
+  async getNotificationInteractionsData(userId) {
+    const interactions = await prisma.notificationInteractions.findMany({
+      where: {
+        userId: userId,
+      },
+      orderBy: {
+        timestamp: "desc",
+      },
+    });
+    return interactions;
+  }
+
+  async getUserMostActiveTimeWIndows(userId) {
```

Comment:
nit: capitalization

Reply:
@azhiah , Sorry, I don't understand. What should I capitalize?

Reply:
I just meant the I in Windows is capitalized when it shouldn't be